runhouse/resources/functionals/mapper.py

```diff
@@ -0,0 +1,137 @@
+import logging
+
+import ray
+
+from runhouse.resources.functions import function, Function
+
+from runhouse.resources.module import Module
+
+logger = logging.getLogger(__name__)
+
+
+class Mapper(Module):
+    def __init__(self, module, method, num_replicas=-1, replicas=None, **kwargs):
+        super().__init__(**kwargs)
+        self.module = module
+        self.method = method
+        self.num_replicas = num_replicas
+        self._auto_replicas = []
+        self._user_replicas = replicas or []
+        self._last_called = 0
+        if self.num_replicas > len(self.replicas) and self.num_replicas > 0:
+            self._add_auto_replicas(self.num_replicas - len(self.replicas))
+
+    @property
+    def replicas(self):
```

Comment:
i like `replicated_modules`, way more clear that these are all `Modules` we're dealing with

Reply:
i'd propagate this name change everywhere

Reply:
I'm leaving it to be consistent with other uses of `replicas`, notably ray serve